druid/src/widget/slider.rs

```diff
@@ -56,6 +58,24 @@ impl Slider {
         self.max = max;
         self
     }
+
+    /// Builder-style method to set the stepping.
+    ///
+    /// The default step size is `0.0` (smooth).
+    pub fn with_step(mut self, step: f64) -> Self {
+        if step < 0.0 {
+            tracing::warn!("bad stepping (must be positive): {}", step);
+            return self;
+        }
+        self.step = if step > 0.0 {
+            Some(step)
+        } else {
+            // A stepping value of 0.0 would yield an infinite amount of steps.
+            // Enforce no stepping instead.
+            None
+        };
```

Comment:
Should we also warn for 0.0 ?

Reply:
I don't think so - see the earlier discussion between Jaap and me.
`0.0` is a valid value, indicating smooth stepping (as opposed to discrete stepping).

Reply:
so `.with_step(0.0)` is a way to clear the previous step if any